ProcessInstanceMigrationClusteredTest.shouldRejectMigrationWhenMessageSubscriptionMigrationIsNotDistributed is flaky

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

<details><summary>Logs</summary>
<pre>
Error:  io.camunda.zeebe.it.clustering.ProcessInstanceMigrationClusteredTest.shouldRejectMigrationWhenMessageSubscriptionMigrationIsNotDistributed -- Time elapsed: 10.83 s <<< FAILURE!
java.lang.AssertionError: 

Expecting Optional to contain a value but it was empty.
	at io.camunda.zeebe.it.clustering.ProcessInstanceMigrationClusteredTest.shouldRejectMigrationWhenMessageSubscriptionMigrationIsNotDistributed(ProcessInstanceMigrationClusteredTest.java:112)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.lang.Thread.run(Thread.java:1583)
</pre>
</details>

[Test results for [IT] Zeebe QA - Integration Tests.zip](https://github.com/user-attachments/files/16026182/Test.results.for.IT.Zeebe.QA.-.Integration.Tests.zip)

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ProcessInstanceMigrationClusteredTest.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ProcessInstanceMigrationClusteredTest.java
@@ -18,7 +18,6 @@ import io.camunda.zeebe.protocol.impl.SubscriptionUtil;
 import io.camunda.zeebe.protocol.record.Assertions;
 import io.camunda.zeebe.protocol.record.RecordType;
 import io.camunda.zeebe.protocol.record.RejectionType;
-import io.camunda.zeebe.protocol.record.intent.CommandDistributionIntent;
 import io.camunda.zeebe.protocol.record.intent.MessageSubscriptionIntent;
 import io.camunda.zeebe.protocol.record.intent.ProcessInstanceMigrationIntent;
 import io.camunda.zeebe.protocol.record.intent.ProcessMessageSubscriptionIntent;
@@ -100,16 +99,7 @@ public class ProcessInstanceMigrationClusteredTest {
     final var secondTargetProcessDefinitionKey =
         deploymentEvent.getProcesses().get(2).getProcessDefinitionKey();
 
-    assertThat(
-            RecordingExporter.commandDistributionRecords(CommandDistributionIntent.ACKNOWLEDGED)
-                .withDistributionPartitionId(2)
-                .findFirst())
-        .isPresent();
-    assertThat(
-            RecordingExporter.commandDistributionRecords(CommandDistributionIntent.ACKNOWLEDGED)
-                .withDistributionPartitionId(3)
-                .findFirst())
-        .isPresent();
+    clientRule.waitUntilDeploymentIsDone(deploymentEvent.getKey());
 
     final int processInstancePartitionId = 2;
     final long processInstanceKey =
@@ -237,16 +227,7 @@ public class ProcessInstanceMigrationClusteredTest {
     final var targetProcessDefinitionKey =
         deploymentEvent.getProcesses().get(1).getProcessDefinitionKey();
 
-    assertThat(
-            RecordingExporter.commandDistributionRecords(CommandDistributionIntent.ACKNOWLEDGED)
-                .withDistributionPartitionId(2)
-                .findFirst())
-        .isPresent();
-    assertThat(
-            RecordingExporter.commandDistributionRecords(CommandDistributionIntent.ACKNOWLEDGED)
-                .withDistributionPartitionId(3)
-                .findFirst())
-        .isPresent();
+    clientRule.waitUntilDeploymentIsDone(deploymentEvent.getKey());
 
     final ProcessInstanceEvent processInstanceEvent =
         clientRule

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ProcessInstanceMigrationClusteredTest.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/ProcessInstanceMigrationClusteredTest.java
@@ -92,8 +92,6 @@ public class ProcessInstanceMigrationClusteredTest {
             .addProcessModel(TARGET_PROCESS_2, "targetProcess2.bpmn")
             .send()
             .join();
-    final var sourceProcessDefinitionKey =
-        deploymentEvent.getProcesses().get(0).getProcessDefinitionKey();
     final var targetProcessDefinitionKey =
         deploymentEvent.getProcesses().get(1).getProcessDefinitionKey();
     final var secondTargetProcessDefinitionKey =

